path-manager/app/services/PathStore.scala

```diff
@@ -125,4 +125,33 @@ object PathStore {
     val paths = pathItems.map{ PathRecord(_) }
     paths.groupBy(_.`type`)
   }
+
+  def putPathItemAndAwaitIndexUpdate(record: PathRecord) {
+
+    Dynamo.pathsTable.putItem(record.asDynamoItem)
+
+    def waitForIndexUpdate(attempts: Int = 0): Boolean = {
+
+      if (attempts > 20) {
+        false
+      } else {
+        val pathRecordsFromIndex = Dynamo.pathsTable.getIndex("id-index").query(new KeyAttribute("identifier", record.identifier), new RangeKeyCondition("type").eq(record.`type`))
+
+        val pathRecordFromIndex = pathRecordsFromIndex.map {
+          PathRecord(_)
+        }.headOption
+
+        pathRecordFromIndex match {
+          case Some(i) if (i.path == record.path) => true
+          case _ => {
+            Thread.sleep(50)
+            waitForIndexUpdate(attempts + 1)
+          }
+        }
+      }
+    }
+
+
+
+  }
```

Comment:
As discussed probably not the cleanest way to achieve write consistency, but pragmatic for doing the job.